### PR TITLE
[MIG] sale_variant_configurator: Migration to 10.0

### PR DESCRIPTION
**Pull request by chienandalu**
Sale - Product variants
=======================

This module allows you to create the product variant when a sale order is
confirmed. It adds to the sale line a product configurator, so that selecting
a product and its attributes can serve for creating/selecting the product
variant.

Depends on:

- [x] product_variant_configurator: https://github.com/OCA/product-variant/pull/65

cc @Tecnativa

### CODE REVIEW

File: sale_variant_configurator/models/sale_order.py
Comment (pedrobaeza) [inline sale_variant_configurator/models/sale_order.py]: Is this still needed? Now everything is in new API and overwriting the field maybe is enough. Please try it (and this is for sale_stock and sale_stock_variant_configurator).

File: sale_variant_configurator/models/sale_order.py
Comment (pedrobaeza) [inline sale_variant_configurator/models/sale_order.py]: You have missed here context argument

File: sale_variant_configurator/models/sale_order.py
Comment (pedrobaeza) [inline sale_variant_configurator/models/sale_order.py]: Why this strange indentation? When lambda is a bit large, better to extract it to a method.

Comment (pedrobaeza) [reply sale_variant_configurator/models/sale_order.py]: And why do you need this?

File: sale_variant_configurator/models/sale_order.py
Comment (pedrobaeza) [inline sale_variant_configurator/models/sale_order.py]: Incorrect indentation. When you have a multi-line `if`, always enclose full expression in parenthesis:
```
if (self.product_tmpl_id and self.order_id.pricelist_id and
        self.order_id.partner_id):
    ...
```

File: sale_variant_configurator/models/sale_order.py
Comment (pedrobaeza) [inline sale_variant_configurator/models/sale_order.py]: Why do you add this? The name is built automatically from template name + attributes from the base module. You are destroying here that names.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by chienandalu** [comment]
Yes, @pedrobaeza, there were some wrong approaches that I've corrected with your suggestions.

**Review by pedrobaeza — APPROVED** [review]
But needs to wait until the dependency is merged to see correct CIs

**Comment by pedrobaeza** [comment]
@chienandalu didn't you include here sale_stock_variant_configurator. It should be, as if not, you get an error.

**Comment by chienandalu** [comment]
@pedrobaeza Should I do it in this same PR or open a new one?

**Comment by pedrobaeza** [comment]
In the same or you will get bad CI. Any way, please wait until the dependent module is merged and you rebase.

**Comment by chienandalu** [comment]
@pedrobaeza Added `sale_stock_variant_configurator`

**Comment by pedrobaeza** [comment]
Can you check Travis status?

**Comment by pedrobaeza** [comment]
Add a temp commit pointing to the branch with product_variant_configurator

**Comment by pedrobaeza** [comment]
I have merge the dependency, so forget it.

File: sale_variant_configurator/models/sale_order.py
Comment (pedrobaeza) [inline sale_variant_configurator/models/sale_order.py]: Are you sure you can remove this? This is for avoiding to lose product_tmpl_id values each time you update the modules.

**Comment by pedrobaeza** [comment]
@luismontalba please review

**Comment by luismontalba** [comment]
Please @chienandalu , make a rebase to able loading of dependences ready merged

**Comment by chienandalu** [comment]
@luismontalba There you have it

**Review by luismontalba — APPROVED** [review]
Tested on runbot

**Comment by pedrobaeza** [comment]
@chienandalu I have checked that variant extra prices are not taken into account right now when selecting attributes with that prices. Can you check?

**Comment by chienandalu** [comment]
@pedrobaeza It seems that only happens when the product variants (`product.product`) doensn't exist. I trying to find out where can I force `price_unit` recalculation in that case.

**Comment by pedrobaeza** [comment]
Yeah, the case is when the variant doesn't exist

**Comment by chienandalu** [comment]
@pedrobaeza Let's see now. I'll try work on more tests.

**Comment by pedrobaeza** [comment]
@chienandalu I have fixed the price extra issue extracting the same code that was in _onchange_product_uom (https://github.com/OCA/product-variant/pull/70/files#diff-fcfba1070c2c0bdc32523ebd31b1f3c1R113), and calling it from 2 sides:

* Changing an attribute (https://github.com/OCA/product-variant/pull/70/files#diff-fcfba1070c2c0bdc32523ebd31b1f3c1R143)
* Changing uom/qty (https://github.com/OCA/product-variant/pull/70/files#diff-fcfba1070c2c0bdc32523ebd31b1f3c1R151)

**Comment by pedrobaeza** [comment]
Finally merging this! :tada:

**Pull request merged**
